modelci/app/v1/endpoints/model.py

```diff
@@ -102,28 +111,121 @@ def update_model_structure_as_new(id: str, structure, dry_run: bool = False):  #
+
+            # check if the provided type matches the original type
+            layer_type = type(net_layer)
+            layer_type_provided = eval(layer_param.type_.value)
+            if layer_type is not layer_type_provided:
+                raise ModelStructureError(f'Expect `{layer_name}.type_` to be {layer_type}, '
+                                          f'but got {layer_type_provided}')
+
+            # get layer parameters
+            layer_param_old = layer_param.parse_layer_obj(net_layer)
+            layer_param_data = layer_param_old.dict(exclude_none=True, exclude={'type_', 'op_'})
+
+            layer_param_update_data = layer_param.dict(exclude_none=True, exclude={'type_', 'op_'})
+            # replace 'null' with None. See reason :class:`ModelLayer`.
+            for k, v in layer_param_update_data.items():
+                if v == 'null':
+                    layer_param_update_data[k] = None
+
+            # update the layer parameters
+            layer_param_data.update(layer_param_update_data)
+            layer = layer_type(**layer_param_data)
+            setattr(net, layer_name, layer)
+
+        elif layer_op == Operation.ADD:
+
+            # check if the layer name not exists
+            if hasattr(net, layer_name):
+                raise ModelStructureError(f'Structure layer name `{layer_name}` found in model {id}.'
+                                          'Operation not permitted.')
+            layer_type = eval(layer_param.type_.value)
+            layer_param_data = layer_param.dict(exclude_none=True, exclude={'layer_type'})
+            layer = layer_type(**layer_param_data)
+            setattr(net, layer_name, layer)
+
+            # change `forward` function
+            raise ValueError('Operation not permitted. Please use `update_model_structure_as_new`.')
+
+        elif layer_op == Operation.DELETE:
+
+            # check if the layer name exists
+            if not hasattr(net, layer_name):
+                raise ModelStructureError(f'Structure layer name `{layer_name}` not found in model {id}.')
+
+            delattr(net, layer_name)
+
+            # change `forward` function
+            raise ValueError('Operation not permitted. Please use `update_model_structure_as_new`.')
+
+    input_tensors = list()
+    for input_ in model.inputs:
+        input_tensor = torch.rand(1, *input_.shape[1:]).type(model_data_type_to_torch(input_.dtype))
+        input_tensors.append(input_tensor)
+
+    # parse output tensors
+    output_shapes = list()
+    output_tensors = net(*input_tensors)
+    if not isinstance(output_tensors, (list, tuple)):
+        output_tensors = (output_tensors,)
+    for output_tensor in output_tensors:
+        output_shape = IOShape(shape=output_tensor.shape, dtype=type_to_data_type(output_tensor.dtype))
+        output_shapes.append(output_shape)
+
+    if not dry_run:
+        register_model(
+            net, dataset='', metric=model.metric, task=model.task,
```

Comment:
Should we set model.metric to empty as the dataset ? As it will be updated during later on fintune process

Reply:
Yes, we should.